packages/medusa/src/services/cart.ts

```diff
@@ -1678,16 +1682,35 @@ class CartService extends TransactionBaseService {
           )
         }
 
-        paymentSession.is_selected = true
+        const sessionInput: PaymentSessionInput = {
+          cart,
+          customer: cart.customer,
+          amount: cart.total!,
+          currency_code: cart.region.currency_code,
+          provider_id: providerId,
+          payment_session_id: paymentSession.id,
+        }
+
+        const isThirdPartyReached =
```

Comment:
comment: I am on the fence about whether or not it makes sense to add a flag for when third-party communication has been done. I don't think we can always rely on `data.length` as an indicator. 

We discussed this shortly @adrien2p. Would like to know what you think @srindom.

Reply:
Yes definitely, what about is_initiated? Which would mean that the createPayment from the plugin had been called. In the new processor api it would be after initiatePayment

Reply:
About the migration, for the current sessions, should we suppose that as soon as there is some data it means that it has been initiated?